pkg/iac/rego/filter.go

```diff
@@ -0,0 +1,85 @@
+package rego
+
+import (
+	"github.com/open-policy-agent/opa/v1/ast"
+
+	"github.com/aquasecurity/go-version/pkg/semver"
+	"github.com/aquasecurity/trivy/pkg/iac/framework"
+	"github.com/aquasecurity/trivy/pkg/log"
+)
+
+type RegoModuleFilter func(module *ast.Module, metadata *StaticMetadata) bool
+
+// TrivyVersionFilter returns a filter that allows only those modules,
+// that are compatible with the given version of Trivy.
+func TrivyVersionFilter(trivyVer string) RegoModuleFilter {
+	if trivyVer == "dev" {
+		return func(_ *ast.Module, _ *StaticMetadata) bool {
+			return true
+		}
+	}
+
+	tv, tverr := semver.Parse(trivyVer)
+	if tverr != nil {
+		log.Warn(
+			"Failed to parse Trivy version - cannot confirm if all modules will work with current version",
+			log.Prefix("rego"),
+			log.String("trivy_version", trivyVer),
+			log.Err(tverr),
+		)
+		return func(_ *ast.Module, _ *StaticMetadata) bool {
+			return true
```

Comment:
There is an outstanding case that I'm not sure if this will affect or not. https://github.com/aquasecurity/trivy-operator/issues/2614

In such a scenario where we cannot determine Trivy version (in the operator it's because Trivy is used as a library and appVersion isn't set), should we filter out?